twitter_pr.rs

```diff
@@ -0,0 +1,96 @@
+extern crate reqwest;
+
+use std::sync::mpsc::{Receiver, channel};
+use serde::{Serialize, Deserialize};
+use tokio::stream::StreamExt;
+use tokio::io::{AsyncBufReadExt, StreamReader};
+use serde_json::{Deserializer, StreamDeserializer};
+use std::io::Write;
+
+pub struct TwitterApi {
+    pub bearer_api_token: String
+}
+
+struct RequestStream {
+    receiver: Receiver<bytes::Bytes>
+}
+
+impl std::io::Read for RequestStream {
+    fn read(&mut self, mut buf: &mut [u8]) -> std::io::Result<usize> {
+        match self.receiver.recv() {
+            Ok(next) => {
+                let vec = next.to_vec();
+                let vec_len = vec.len();
+                println!("Incoming -> {}", String::from_utf8(vec.clone()).unwrap_or("err".to_string()));
+                buf.write_all(vec.as_slice());
```

Comment:
You should handle the possible error from write_all. You can use the `?` operator since it's an `io::Error`.

Also, IMO it's more idiomatic to use `&` to get vec as a slice:

```suggestion
                buf.write_all(&vec)?;
```

By the way, have you considered what you want to do when `vec` is bigger than `buf`? Since `buf` is a slice, not a vec, it can't be resized. One option would be for `RequestStream` to store whatever bytes went unread, and return them in future `.read()` calls. Or you could just punt and call that an error.

Also, I think it's possible you might receive a `Vec` here that has zero length. That would cause you to return 0 from `.read()`, which means EOF. You should handle that case specially - for instance, by calling `recv()` again (unless it's actually time to end the stream).